automata4j/src/main/java/com/avrsandbox/fsa/core/TransitionalManager.java

```diff
@@ -88,17 +97,33 @@ public <I, O> void assignNextState(final Transition<AutoState<I, O>> transition)
     }
     
     /**
-     * Transits to the next-state from a state-transitionPath.
+     * Traverses through a transition path starting from the present-state
+     * assigning the next-state without transiting to it.
+     *
+     * <p>
+     * Tips: A call to {@link TransitionalManager#transit(TransitionListener)} can be submitted from the
+     * {@link TransitionListener#onTransition(AutoState)} to transit to the assigned next-state.
+     * </p>
      * 
      * @param <I> the state input type
      * @param <O> the state tracer object type
-     * @param transitionPath the system state-transitionPath holding a presentstate and a nextstate
+     * @param transitionPath the system state-transitionPath holding a present state and a next state
      * @param transitionListener an event driven interface object that fires {@link TransitionListener#onTransition(AutoState)} 
-     *                           after the {@link AutoState#invoke(Object)} is invoked when the transition completes
+     *                           after the {@link AutoState#invoke(Object)} is invoked when the transition to the present-state completes
      */
     public <I, O> void transit(final TransitionPath<AutoState<I, O>> transitionPath, final TransitionListener transitionListener) {
-        assignNextState(transitionPath.getNextState());
-        transit(transitionPath.getNextState().getInput(), transitionListener);
+        assignNextState(transitionPath.getPresentState());
+        transit(transitionPath.getPresentState().getInput(), new TransitionListener() {
+            @Override
+            @SuppressWarnings("all")
+            public <I, O> void onTransition(AutoState<I, O> presentState) {
```

Comment:
@sonatype-lift ignore

Reply:
This is **not a critical type shadowing** because this is not a user code, so no operations are performed using these type parameters and they got erased to the super-type `Object` in compile time.